### PR TITLE
Feature/update deps and ts

### PR DESCRIPTION
**Pull request by toloudis**
Beginning to pull this codebase toward modern typescript.
Many files changed but nothing too huge.

* Update almost all the dev dependencies
* Add typescript support in the linter and babel parsers
* Convert a couple of files to typescript as a test
* Make sure the test app still runs
* remove old gradle and Jenkins files that are no longer used
* eslint is now only linting the src/ directory for the moment, and there are so many warnings I am running it in quiet mode for now.  Future changes will fix all lint warnings.
* ViveController had to be converted to a class but most of the changes there are just indenting/whitespace

### CODE REVIEW

File: src/Histogram.ts [hunk 1]
```diff
@@ -1,12 +1,8 @@
 import { getColorByChannelIndex } from "./constants/colors.js";
 
-(function() {
-  if (!Math.clamp) {
-    Math.clamp = function(val, cmin, cmax) {
-      return Math.min(Math.max(cmin, val), cmax);
-    };
-  }
-})();
+function clamp(val: number, cmin: number, cmax: number): number {
```
Comment (EricJIsaac) [inline src/Histogram.ts:3]: Do the Math.clamp -> clamp changes correspond to a js vs. ts difference? (were they needed to pass the linter, or changed for another reason)

**Review by EricJIsaac — APPROVED** [review]

File: src/Histogram.ts (hunk 1 above)
Comment (toloudis) [reply src/Histogram.ts:3]: Yes I think typescript was complaining but also, I have always felt bad about monkeypatching the global Math object.  It's really not great practice.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,16 +1,57 @@
...
   parserOptions: {
-    ecmaVersion: "2018",
+    project: "./tsconfig.json",
+    ecmaVersion: "2020",
     sourceType: "module",
   },
+  plugins: ["@typescript-eslint", "react"],
   rules: {
+    "@typescript-eslint/ban-types": ["warn"],
+    "@typescript-eslint/naming-convention": [
```
Comment (schoinh) [inline .eslintrc.js:17]: Ooh, I didn't know you could lint for this

File: README.md [hunk 3]
```diff
@@ -58,16 +58,6 @@ view3D.updateLuts(aimg);
...
-    rules: [
-        ...,
-        {
-            test: /Worker\.js$/,
-            use: 'worker-loader?inline=true'
-        }
-    ]
-```
 - in `VolumeViewer.jsx`
 ```JavaScript
```
Comment (schoinh) [inline README.md:62]: I know this is not part of this PR but I just noticed that this code block is not closed properly. The backticks in line 131 need to be un-indented. 

image

File: src/test/lut.test.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
 import { expect } from "chai";
 
-import Histogram from "../Histogram.js";
+import Histogram from "../Histogram.ts";
 import VolumeMaker from "../VolumeMaker.js";
 
+function clamp(val, cmin, cmax) {
+  return Math.min(Math.max(cmin, val), cmax);
+}
+
```
Comment (schoinh) [inline src/test/lut.test.js:9]: Maybe throw this in a util file if it's going to be used in multiple places (here and in Histogram.ts)?

**Review by schoinh — APPROVED** [review]
This must feel good to do...

File: README.md (hunk 3 above)
Comment (toloudis) [reply README.md:62]: Thanks! will fix that

File: src/test/lut.test.js (hunk 4 above)
Comment (toloudis) [reply src/test/lut.test.js:9]: The only reason I didn't is that the repeated use is in a unit test.  One more repetition and I will refactor it :)

**Comment by toloudis** [comment]
> This must feel good to do...

It'll feel even better to convert the whole thing to typescript (but at the same time I dread how long it will take)

**Pull request merged**